cpp/src/arrow/util/ree_util_test.cc

```diff
@@ -101,7 +101,7 @@ TYPED_TEST_P(ReeUtilTest, PhysicalLength) {
   ASSERT_EQ(internal::FindPhysicalLength(run_ends246, 4, 0, 7), 0);
 }
 
-TYPED_TEST_P(ReeUtilTest, MergedRunsInterator) {
+TYPED_TEST_P(ReeUtilTest, MergedRunsIteratorTest) {
```

Comment:
Nope, that fails because there's *already* a [`MergedRunsIterator`](https://github.com/apache/arrow/blob/2ea7f79541f2024ef41aa823fd8ad2603dc2b6f8/cpp/src/arrow/util/ree_util.h#L437)

Trust me that if there's a simple fix, I will go with it. I only go w/ complicated things when the obvious correction fails CI or similar.

Reply:
Hmm, fair enough.